### DISCUSSION TITLE
streams-audiokit-serial not working, but input.ino does?

### DISCUSSION THREAD

**Discussion by mmcc-xx** [pschatzmann/arduino-audio-tools#828] [Q&A] [answered]
I got an ESP32-Audio-Kit from Amazon. Photo below - note the version number is V2.2 A237.

I tried to get the example streams-audiokit-serial working, but the output is a continuous stream of 0, 0. I tried  AUDIOKIT_BOARD values of 5, 6, 7, 10 and 11 in AudioKitSettings.h. 5 and 10 give the stream of zeros, and the other ones don't work at all. I also tried AUDIO_HAL_ADC_INPUT_LINE1, AUDIO_HAL_ADC_INPUT_LINE2, and AUDIO_HAL_ADC_INPUT_LINE3 as the input device with the same results.

So I then tried the input.ino example from the arduino-audiokit repository. This works fine with AUDIOKIT_BOARD = 5 and the input device set to AUDIO_HAL_ADC_INPUT_LINE2.

Any suggestions for how to fix this? I'd like to get streams-audiokit-serial working so I can move on to streaming the output over http.

![image_67229697](https://user-images.githubusercontent.com/1315341/236697473-53d8b298-16bd-41aa-bdfc-25cadcceb248.JPG)

**Comment by pschatzmann** [comment, marked as answer]
Did you try the sine output test ?
Can it be that you installed an version that does not contain this correction ? https://github.com/pschatzmann/arduino-audio-tools/commit/40d998ade83bf8b5b2422efa36efb26cc4735eda

**Reply by mmcc-xx** [reply]
Updating the arduino-audio-tools library fixed it. Thanks!
